File: 4. Median of Two Sorted Arrays/Median of Two Sorted Arrays v2.js
// FAIL

const nums1 = [1,3]
const nums2 = [2]

const findMedianSortedArrays = (nums1,nums2) => {
    let nums_concat = nums1.concat(nums2)
    let output = []
    const len_arr = nums_concat.length
    
    // while ( x < y)
    
    for (let l = 0; l < len_arr; ++l) {
        console.log(`> ${l}`)
        for (let r = 0; r < len_arr; ++r) {
            console.log(`==> ${r}`)
            if ( nums_concat[l] > nums_concat[r]) {
                console.log(`====> ${nums_concat[l]} > ${nums_concat[r]}`)
                output.splice(nums_concat[l])
                console.log(`====> push ${nums_concat[l]}`)
            } else if (nums_concat[l] < nums_concat[r]) {
                output.splice(r,1,nums_concat[l])
                console.log(`----> ${nums_concat[l]} < ${nums_concat[r]}`)
            }
            console.log(`++++> ${output}`)

        }
    }
};

findMedianSortedArrays(nums1,nums2)

// nums2.splice(0,0,1)
// nums2.splice(5,0,33)
// nums2.splice(2,0,5)
// nums2.splice(2,1,33)

// console.log(nums2)
// console.log(nums1+nums2)

File: 4. Median of Two Sorted Arrays/Median of Two Sorted Arrays v2.go
package main

import "fmt"

func findMedianSortedArrays2(nums1 []int, nums2 []int) float64 {
    
	var output float64

	arr := append(nums1,nums2...)
	// arr := []string{"a","b","c"}
	for {
		sorted := true
		fmt.Println("sebelum sort =>", arr)

		// for i,x := range arr {
		for i:=0; i < len(arr)-1; i++{
			if arr[i] > arr[i+1] {
				temp := arr[i]
				arr[i] = arr[i+1]
				arr[i+1] = temp
				sorted = false
			}
		}
		fmt.Println("Sesudah sort => ", arr)
		if sorted == true {break}
	}

	odd := len(arr)
	indeks := (odd/2)
	if odd%2 == 0 {
		output = ((float64(arr[indeks])+float64(arr[indeks-1]))/2)
	} else {
		output = float64(arr[indeks])
	}
	// fmt.Println(odd)

	return output
}